mlir/lib/Target/LLVMIR/DebugImporter.cpp

```diff
@@ -85,10 +85,15 @@ DIDerivedTypeAttr DebugImporter::translateImpl(llvm::DIDerivedType *node) {
   DITypeAttr baseType = translate(node->getBaseType());
   if (node->getBaseType() && !baseType)
     return nullptr;
+  DINodeAttr extraData =
+      translate(llvm::dyn_cast_or_null<llvm::DINode>(node->getExtraData()));
+  if (node->getExtraData() && !extraData)
+    mlirModule.emitWarning(
+        "dropped DIDerivedType's extra data due to having an unsupported type");
```

Comment:
nit: I would probably drop the warning since there are many fields that are currently not supported and adding a warning for everything would be noisy and not very informative.